transcom/mymove
# B 19997 INT - V2

## [B-19997](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A949600)
## [Original Integration Review](https://github.com/transcom/mymove/pull/12910)
## Summary

Initial Review:
"This task creates the standalone shipment functionality for the office app, and the UI for the TIO role. When a standalone shipment is part of a payment request, it will be displayed with a '- Standalone' tag on it to alert the TIO. Additionally, when they are reviewing the payment request it will be noted there that the shipment is a standalone, and in the calculation details it will display the Standalone Payment Cap, which is currently 1000$. Any Standalone Domestic Crating service item should not exceed this cap regardless of how large the crate is."

New review:
Additionally, the 'show calculations' button when reviewing this shipment should display the 'Uncapped Request Total'. This is the total as calculated before the cap was applied.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Setup: Run db_dev_fresh to ensure db changes are in place

Note: When testing I found that many of the service items for existing data are not complete, and will cause the creation of a payment request to error out. To test a payment request with multiple service items it is easiest to just create them yourself.

1. Access the Office app
2. Login to the Prime Sim
3. Add a domestic crating service item with the standalone box checked.

4. Login as a TOO
5. Find the Move you edited in the Prime Sim
6. Approve the newly created Service Item

5. Login to the Prime Sim
6. Navigate to the previously edited move
7. Create a payment request that includes the Standalone crating item you created

8. Login as a TIO
9. Find the move you edited in the Prime Sim
10. Navigate to the payment requests tab
11. Note that the Domestic crate service item you created has '- Standalone' after domestic crating
12. Review the payment request
13. Note that the Domestic crate service item you created has '- Standalone' after domestic crating in the payment request review
14. Click on calculations to view the calculations
15. Note that the Uncapped Request Total is displayed
16. Note that the standalone payment cap is displayed

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
![Screenshot 2024-05-31 at 10 03 46 AM](https://github.com/transcom/mymove/assets/162157577/61bf5873-09c8-4f76-8882-7b3da478ce7b)

![Screenshot 2024-06-04 at 2 08 24 PM](https://github.com/transcom/mymove/assets/162157577/6c55a8b2-0c19-4810-9c89-5430671b44fa)
